Adapt modal design to large screens

RasmusKjeldgaard opened the pull request:
## Which issue does this PR close?

This PR closes #2977

## What is the new behavior?

<!-- Replace this paragraph with a description of the new behaviour after your pull request is merged -->

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this PR contains a breaking change, replace this paragraph with a description of the impact and migration path for existing applications  -->

## Are there any additional context?

<!-- Replace this paragraph with any additional context e.g, explanations, links or screenshots (if any) -->

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

RasmusKjeldgaard committed:
Adapt modal wrapper to larger screen sizes

RasmusKjeldgaard committed:
Add sm, md and lg sizes to modal on large screens

RasmusKjeldgaard committed:
Use existing (unused) ModalSize for non-fullscreen sizes

RasmusKjeldgaard committed:
Make drawer and modal header buttons similar

RasmusKjeldgaard committed:
Remove shared example scss from modal example

It gives unwanted padding on page-based examples, as e.g. pages
scrollbar with be spaced from the viewport edge.

RasmusKjeldgaard committed:
Make drawer and modal medium sized by default

This change only affects desktop, as sizes are not implemented for mobile

RasmusKjeldgaard committed:
Adjust design of small, medium and large modal on larger screens

RasmusKjeldgaard committed:
WIP: Update showcase examples with size selector

RasmusKjeldgaard committed:
Adjust modal title and toolbar paddings

RasmusKjeldgaard committed:
Fix spacings of toolbar, footer and collapsible title

RasmusKjeldgaard committed:
Make icon in close button in header comsistent across flavors

RasmusKjeldgaard committed:
Do not show footer in examples by default

RasmusKjeldgaard committed:
Align modal wrapper tests to new design

RasmusKjeldgaard committed:
Use dropdown for size selection in modal config

RasmusKjeldgaard committed:
Make no button decoration for custom interact-with-background behavior

RasmusKjeldgaard committed:
Align modals tests to design changes

RasmusKjeldgaard committed:
Preserve interact-with-background drawer functionality on desktop

RasmusKjeldgaard committed:
Temporarily remove flaky modal helper test suite

RasmusKjeldgaard committed:
Fix failing tests

RasmusKjeldgaard committed:
Make header border appear on scroll on mobile

RasmusKjeldgaard committed:
Remove magin on buttons in modal header and footer

This is to avoid magic numbers for paddings in code and tests
as the margin would have to be subtracted from the paddings in
header and footer for the design to look correct.

henrikvoetmand commented:
![modal-drawer-review](https://user-images.githubusercontent.com/71710012/234579420-1f5da2fe-1acd-4f37-a5d3-eb200954a991.png)

RasmusKjeldgaard committed:
Adjust paddings for correct header height

RasmusKjeldgaard committed:
Move snapFooterToKeyboard to example config

RasmusKjeldgaard committed:
Revert whitespace change

RasmusKjeldgaard committed:
Remove unused imports

RasmusKjeldgaard committed:
Reorder modal size variables

RasmusKjeldgaard committed:
Remove unnecessary modal shadow margin

RasmusKjeldgaard committed:
Refactor global styles for readability

RasmusKjeldgaard committed:
Simplify modal wrapper button usage

RasmusKjeldgaard committed:
Refactor scroll logic to be more responsive

Includes logic to run outside zone to avoid extensive amount of CD
cycles while still updating the property inside the zone to sync it
properly to the template

RasmusKjeldgaard committed:
Cleanup modal wrapper

RasmusKjeldgaard committed:
Add support for 'height' in ModalConfig

RasmusKjeldgaard committed:
Add back full-height functionality needed on mobile drawer

RasmusKjeldgaard committed:
Enable height config on iphone component

RasmusKjeldgaard committed:
Unify default example to simplify showcase

RasmusKjeldgaard committed:
Remove duplicate test

RasmusKjeldgaard committed:
Contain content for drawers on mobile only

RasmusKjeldgaard commented on libs/designsystem/modal/src/modal-wrapper/config/modal-config.ts:
```diff
@@ -10,6 +10,7 @@ export interface ModalConfig {
   collapseTitle?: boolean;
   component: any;
   size?: ModalSize;
+  height?: string;
```
```suggestion
  customHeight?: string;
```
Rename to `customHeight` as we want it to be clear that this is something that _overrides_ size

RasmusKjeldgaard committed:
Simplify bottom-border styles and make mobile-first

RasmusKjeldgaard committed:
Update header button slots

Page progress should appear before secondary action

RasmusKjeldgaard committed:
Add custom height to modal before presenting

RasmusKjeldgaard committed:
Remove unnecessary style rules for header in collapsible title

RasmusKjeldgaard committed:
Always add page progress to start slot

RasmusKjeldgaard committed:
Use class binding instead of HostBinding for scroll divider to work

RasmusKjeldgaard committed:
Add modal sizes example

RasmusKjeldgaard committed:
Remove todo and reword ionScroll comment

RasmusKjeldgaard committed:
Simplify config in examples and align design

RasmusKjeldgaard committed:
Update size example with customHeight button

RasmusKjeldgaard committed:
Merge branch 'develop' into enhancement/2977-add-large-screen-functionality-to-modal

RasmusKjeldgaard committed:
change height prop in modalconfig to customHeight

RasmusKjeldgaard committed:
Improve modal config

RasmusKjeldgaard committed:
Update device example view

RasmusKjeldgaard committed:
Refactor examples to better fit new showcase

RasmusKjeldgaard committed:
Rework showcase for readability

RasmusKjeldgaard committed:
Merge branch 'develop' into enhancement/2977-add-large-screen-functionality-to-modal

RasmusKjeldgaard committed:
Fix scrollto link

RasmusKjeldgaard committed:
Hide Modal v2 docs

With this modal update the Modal v2 work is postponed

RasmusKjeldgaard committed:
Clean up configuration of components

RasmusKjeldgaard committed:
Rename examples

RasmusKjeldgaard committed:
Allow override for border with modal css class

RasmusKjeldgaard committed:
Merge branch 'develop' into enhancement/2977-add-large-screen-functionality-to-modal

RasmusKjeldgaard committed:
Rename toggle config function

RasmusKjeldgaard committed:
Merge branch 'develop' into enhancement/2977-add-large-screen-functionality-to-modal

RasmusKjeldgaard committed:
Add explaining comment  to page progress margin

RasmusKjeldgaard committed:
Improve readability

RasmusKjeldgaard committed:
Refactor test for readability

RasmusKjeldgaard commented on apps/cookbook/src/app/examples/example-configuration-wrapper/example-configuration-wrapper.component.scss:
```diff
@@ -1,7 +1,47 @@
+      transition-property: transform, border-radius;
+      transition-duration: utils.get-transition-duration('short');
+      transition-timing-function: utils.get-transition-easing('enter-exit');
+      transform: rotate(-90deg) translateY(-#{utils.$border-radius + utils.size('xs')});
```
Restore previous functionality with no rotation and 40% transform on larger screens

RasmusKjeldgaard commented on apps/cookbook/src/app/examples/example-configuration-wrapper/example-configuration-wrapper.component.html:
```diff
@@ -1,4 +1,15 @@
-<ng-container *ngIf="snapToViewport; else fieldset">
+<button
```
For line 20: `<legend>Configuration</legend>` should be configurable via input prop but default to 'Configuration'

RasmusKjeldgaard committed:
Reestablish css selector

RasmusKjeldgaard committed:
Finetune showcase

RasmusKjeldgaard committed:
Add back desktop config wrapper styles

RasmusKjeldgaard committed:
Set defaults for nested config options

RasmusKjeldgaard committed:
Allow setting title on config wrapper

RasmusKjeldgaard committed:
Merge branch 'develop' into enhancement/2977-add-large-screen-functionality-to-modal

RasmusKjeldgaard committed:
Use new value on viewMode in header

RasmusKjeldgaard committed:
Rename modal examples

RasmusKjeldgaard committed:
Remove unneeded example scss file

jakobe committed:
docs: refactor modal size selector to own comp

jakobe committed:
docs: re-style config wrapper

jakobe committed:
docs: use toggle-mode for config wrapper

jakobe committed:
docs: remove modal size from modal config

jakobe committed:
docs: remove redundant host selector

jakobe committed:
Merge branch 'enhancement/2977-add-large-screen-functionality-to-modal' of https://github.com/kirbydesign/designsystem into enhancement/2977-add-large-screen-functionality-to-modal

RasmusKjeldgaard commented on apps/cookbook/src/app/examples/modal-example/embedded-modal-example/embedded-modal-example.component.html:
```diff
@@ -32,7 +32,9 @@ <h3>Open nested:</h3>
-      <cookbook-example-configuration-wrapper>
+      <details>
+        <summary>Configuration for nested modal</summary>
+        <!-- <cookbook-example-configuration-wrapper [title]="'Configuration for nested modal'"> -->
```
```suggestion

```

RasmusKjeldgaard commented on apps/cookbook/src/app/examples/modal-example/embedded-modal-example/embedded-modal-example.component.html:
```diff
@@ -32,7 +32,9 @@ <h3>Open nested:</h3>
         Action sheet
       </button>
-      <cookbook-example-configuration-wrapper>
+      <details>
+        <summary>Configuration for nested modal</summary>
```
I love this one! 🙌

RasmusKjeldgaard reviewed (commented):
Just reviewed the added docs stuff @jakobe, looks great!

jakobe committed:
docs: remove commented code

Co-authored-by: RasmusKjeldgaard <rkk@bankdata.dk>

RasmusKjeldgaard committed:
Merge branch 'develop' into enhancement/2977-add-large-screen-functionality-to-modal

RasmusKjeldgaard committed:
Align small modal size to new design spec

RasmusKjeldgaard committed:
Merge branch 'develop' into enhancement/2977-add-large-screen-functionality-to-modal

jakobe reviewed (approved):
:shipit: 🤞🏻

jakobe committed:
Merge branch 'develop' into enhancement/2977-add-large-screen-functionality-to-modal
